srv/service.js

```diff
@@ -1,9 +1,51 @@
-const { buildNotification, messages } = require("./../lib/utils")
+const { buildNotification, messages, validateNotificationTypes, _config_to_ms, calcTimeMs, getNotificationTypesKeyWithPrefix, getPrefix } = require("./../lib/utils")
+const { validateAndPreprocessHandler } = require("../lib/deployer/notificationTypes")
 const cds = require('@sap/cds')
 const LOG = cds.log('notifications');
 
+const DEL_TIMEOUT = {
+  get value() {
+      const timeout_ms = _config_to_ms('notifications_deletion_timeout', '30d')
+      Object.defineProperty(DEL_TIMEOUT, 'value', { value: timeout_ms })
+      return timeout_ms
+  }
+}
+
 class NotificationService extends cds.Service {
+  async init() {
+
+    this.on("*", async (req, next) => {
+      if (req.event === 'deleteNotifications') return next()
+      LOG._debug && LOG.debug('Handling notification event:', req.event);
+      let notificationsToCreate = [];
```

Comment:
What is the use case of `alert.emit('deleteNotifications', {})`. I'm not aware of any reason why we should archiving notifications can you elaborate on this?

Reply:
Saving notifications is done so a 'MinTimeBetweenNotifications' can be specified avoiding to many notifications. But to do these checks notifications have to be saved. I did not yet find a clean way for scheduling a job to trigger deletion automatically, that is why the event exists so apps can trigger it themselves via a scheduler. I thought about adding a HANA Cloud Task Scheduler, but that would't not work for pg/sqlite.

Reply:
I have one use case where I send an action to multiple people. 
If one person actions it, then I can delete the notification for the rest of the recipients.

Reply:
Hi @TheoExall, I think the event does not yet allow to delete specific notifications. However please also consider that Notification actions is something that ideally should be avoided. That is also why there likely won't be any support coming for NotificationType actions. The preferred way is to use SAP Task Center and create a task assigned to multiple people, with them receiving the notification for the task.

Reply:
For task scheduling you can use the event-queue package. This is exactly the purpose of the package.
But however, I still would not add such functionality in a reuse package. For me this seems more application-specific. Because we handle such things in our business coding and would not want additional tables for such things...

Reply:
@schiwekM 
Correct. I have a really small app that is used by IT staff. In my use case there is around 80 notifications per month with around 40 recipients. In this case it's more about cleaning up older notifications, for managing the tasks I agree that Task Center is the best solution

Reply:
> For task scheduling you can use the event-queue package. This is exactly the purpose of the package. But however, I still would not add such functionality in a reuse package. For me this seems more application-specific. Because we handle such things in our business coding and would not want additional tables for such things...

Thanks for mentioning the package - I am just unsure if it is fine to add it as an optional dependency. The tables / functionality is optional. If you do not import the index.cds no table will be added & notifications will be saved.